### PR TITLE
feat: a11y updates for expand/collapse elements

### PR DESCRIPTION
**Pull request by alexandervalencia**
### Description

This ticket supports [API-1252](https://vajira.max.gov/browse/API-1252). This updates expand/collapse sections for operations and schemas to have:
- proper indications of state (up arrow/down arrow point the right way when open/close)
- tab-able buttons for keyboard navigation to open/close sections

### How Has This Been Tested?
Tested using VoiceOver for Mac (native software) and Keyboard navigation

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [X] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [X] are not breaking changes.

### Documentation
- [X] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [X] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/model-collapse.jsx [hunk 1]
```diff
@@ -86,15 +86,13 @@ export default class ModelCollapse extends Component {
 
     return (
       <span className={classes || ""} ref={this.onLoad}>
-        { title && <span onClick={this.toggleCollapsed} className="pointer">{title}</span> }
-        <span onClick={ this.toggleCollapsed } className="pointer">
+        <button aria-expanded={this.state.expanded} className="model-box-control" onClick={this.toggleCollapsed}>
```
Comment (gileswells) [inline src/core/components/model-collapse.jsx:89]: You have some failing unit tests here and it appears like your removal of the `pointer` class is the culprit. The test is searching for that specific class to trigger a click and test expand and collapse but can't find it without the class.

https://github.com/department-of-veterans-affairs/swagger-ui/runs/2324060183?check_suite_focus=true#step:6:1883

Adding a class of `pointer` to the button would sidestep this and fix 2 of the 4 failing tests. It might however be better to update the unit tests though to better target the button itself rather than looking for a `pointer` class that might not even be necessary anymore with the button wrapping everything.

With all the spans being wrapped in a button I think that [this last test](https://github.com/department-of-veterans-affairs/swagger-ui/blob/ft/a11y-updates-for-expand-collapse/test/e2e-cypress/tests/features/model-collapse.js#L50) is going to find itself unnecessary.

Comment (gileswells) [reply src/core/components/model-collapse.jsx:89]: Something to consider here might also be the impacts on any popular swagger themes that could exist. I wasn't able to find a centralized list of themes to consider testing with.

File: src/core/components/models.jsx [hunk 2]
```diff
@@ -58,11 +58,17 @@ export default class Models extends Component {
...
-        </svg>
+      <h4>
+        <button
+          aria-expanded={showModels}
+          className="models-control"
+          onClick={() => layoutActions.show(specPathBase, !showModels)}
+        >
+          <span>{isOAS3 ? "Schemas" : "Models"}</span>
+          <svg width="20" height="20" aria-hidden="true" focusable="false">
+            <use xlinkHref={showModels ? "#large-arrow-up" : "#large-arrow-down"} />
```
Comment (gileswells) [inline src/core/components/models.jsx:69]: The change in names of the arrows throws me off. Without anything else to show that the `large-arrow-down` is in fact pointing up in this instance (like a css rotation) and moving the down arrow to the false condition, this change feels confusing to me.

**Review by gileswells — CHANGES_REQUESTED** [review]
Some failing unit test and a question about arrows.

File: src/core/components/models.jsx (hunk 2 above)
Comment (alexandervalencia) [reply src/core/components/models.jsx:69]: It's not that the icons just changed names... I added a separate svg for an arrow pointing up. 
Previously the collapsed / expanded states were like this:
Collapsed: ▶️
Expanded: 🔽 

Now the states are
Collapsed: 🔽
Expanded: 🔼 

Per my conversation with @brentschnieder, the icons should indicate the intent of the action... in this case, clicking on the button will move content in the direction indicated. If the content needs to come down (expand) the button should show a down-arrow. Clicking on the button changes the icon to an up arrow, indicating that clicking it will move the content back up (collapse). Previously, the icons had no meaningful indication to fulfill a11y requirements

See screenshots for visual details.

Screen Shot 2021-04-12 at 1 03 14 PM
Screen Shot 2021-04-12 at 1 03 23 PM

File: src/core/components/model-collapse.jsx (hunk 1 above)
Comment (alexandervalencia) [reply src/core/components/model-collapse.jsx:89]: The class being used in some themes to control the title was something I hadn't considered, thanks! I added back to the span where the title is being displayed

**Review by gileswells — APPROVED** [review]
LGTM
